### ISSUE TITLE
"Reopen all repositories from last session" always applied?

### ISSUE THREAD

**Issue by namxam** [laullon/gitx#127]
Hi,

I am using OS Lion and GitX always opens repositories from the previous session although I did not check that option in the preferences. It started when I upgraded to Lion and remained that way even after updating to the latest version. I also tried to enabled and disable it again, but it did not help.

Anyone else having the same issue?

**Comment by Uncommon** [comment]
This is because Lion always tells document-based applications to reopen everything that was open when they last quit. GitX has no idea that these open requests are coming from Lion and not from you.

**Comment by namxam** [comment]
Ah, good to know and I guess it was introduced with the new application states, so the system can shut down, reboot, and still reopen everything from the last session.

But it would be great if there was a work around as I switch between projects quite often and do not like a cluttered workspace with 5 gitx windows fighting for my attention ;)

**Comment by Uncommon** [comment]
Yes, it is possible for an application to opt out of the auto-reopen feature.
